Lib/test/test_base64.py

```diff
@@ -1,11 +1,29 @@
 import unittest
 import base64
 import binascii
+import string
 import os
 from array import array
 from test.support import os_helper
 from test.support import script_helper
 
+from test.support.hypothesis_helper import hypothesis
+
+
+@hypothesis.strategies.composite
+def altchars(draw):
+    """Generate 'altchars' for base64 encoding.
+
+    Via https://docs.python.org/3/library/base64.html#base64.b64encode :
+
+    "Optional *altchars* must be a :term:`bytes-like object` of length 2 which
+    specifies an alternative alphabet for the ``+`` and ``/`` characters."
+    """
+    reserved_chars = (string.digits + string.ascii_letters + "=").encode()
+    allowed_chars = hypothesis.strategies.sampled_from(
+        [n for n in range(256) if n not in reserved_chars])
+    return bytes(draw(hypothesis.strategies.lists(allowed_chars, min_size=2,
+                                                  max_size=2, unique=True)))
```

Comment:
Instead of using `@composite` here, we can simply return a strategy from our function:

```suggestion
    return hypothesis.strategies.lists(allowed_chars, min_size=2, max_size=2, unique=True))).map(bytes)
```

In more complicated scenarios, this can be a decent performance improvement (by amortizing the construction and validation over multiple draws, instead of executing the body each time we draw a value) - particularly if we `@functools.cache` the function, or happen to draw from it multiple times (e.g. `st.lists(altchars())`).  Mostly though it's just an idiom to remind readers that strategies are just python values, not magic.